sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,432 @@
+                        row
+                    ]
+                else:
+                    self._log.debug("Columns found for an unkown table with the object_id: {}".format(table_id))
+            else:
+                self._log.debug("Couldn't find id of a table: {}".format(table_name))
+        return len(data)
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_partitions_data(self, table_ids, table_id_to_table_data, cursor):
+        rows = execute_query(PARTITIONS_QUERY.format(table_ids), cursor)
+        for row in rows:
+            table_id = row.pop("id", None)
+            if table_id is not None:
+                table_id_str = str(table_id)
+                if table_id_str in table_id_to_table_data:
+                    table_id_to_table_data[table_id_str]["partitions"] = row
+                else:
+                    self._log.debug("Partition found for an unkown table with the object_id: {}".format(table_id_str))
+            else:
+                self._log.debug("Return rows of [{}] query should have id column".format(PARTITIONS_QUERY))
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_index_data(self, table_ids, table_id_to_table_data, cursor):
+        rows = execute_query(INDEX_QUERY.format(table_ids), cursor)
+        for row in rows:
+            table_id = row.pop("id", None)
+            if table_id is not None:
+                table_id_str = str(table_id)
+                if table_id_str in table_id_to_table_data:
+                    table_id_to_table_data[table_id_str].setdefault("indexes", [])
+                    table_id_to_table_data[table_id_str]["indexes"].append(row)
+                else:
+                    self._log.debug("Index found for an unkown table with the object_id: {}".format(table_id_str))
+            else:
+                self._log.debug("Return rows of [{}] query should have id column".format(INDEX_QUERY))
+
+    @tracked_method(agent_check_getter=agent_check_getter, track_result_length=True)
+    def _populate_with_foreign_keys_data(self, table_ids, table_id_to_table_data, cursor):
+        rows = execute_query(FOREIGN_KEY_QUERY.format(table_ids), cursor)
+        for row in rows:
+            table_id = row.pop("id", None)
+            if id is not None:
+                table_id_str = str(table_id)
+                if table_id_str in table_id_to_table_data:
+                    table_id_to_table_data.get(table_id_str).setdefault("foreign_keys", [])
+                    table_id_to_table_data.get(table_id_str)["foreign_keys"].append(row)
+                else:
+                    self._log.debug("Foreign key found for an unkown table with the object_id: {}".format(table_id_str))
+            else:
+                self._log.debug("Return rows of [{}] query should have id column".format(FOREIGN_KEY_QUERY))
```

Comment:
we shouldn't be logging the entire `FOREIGN_KEY_QUERY` here. let's just log the name of the queyr

Reply:
I removed this log. 'id' column is a part of the query so if we don't get it save to throw as everything is broken.